json/src/spec/builtin.rs

```diff
@@ -57,6 +57,41 @@ pub struct AltBn128Pairing {
 	pub pair: u64,
 }
 
+
+/// Bls12 pairing price
+#[derive(Debug, PartialEq, Deserialize, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct Bls12PairingPrice {
+	/// Price per final exp
+	pub base: u64,
+	/// Price per pair (Miller loop)
+	pub pair: u64,
+}
+
+/// Pricing for constant Bls12 operations (ADD and MUL in G1 and G2, as well as mappings)
+#[derive(Debug, PartialEq, Deserialize, Clone)]
+#[serde(deny_unknown_fields)]
+pub struct Bls12ConstOperationsPrice {
```

Comment:
Perhaps rename to `Bls12ConstOperations` for consistency with the structs immediately above this one (which don't have `Price` as a suffix).